printing_templates/One Dive.html

```diff
@@ -111,117 +148,135 @@
-									<p> {{ dive.duration }} </p>
+									<p> {{ dive.location }} </p>
 								</td>
-							</tr>
-						</tbody></table>
-						<table class="table_class">
-							<tbody><tr>
 								<td class="fieldTitle">
-									<h1> Time. </h1>
+									<b> Water Temp. </b>
 								</td>
 								<td class="fieldData">
-									<p> {{ dive.time }} </p>
-								</td>
+									<p> {{ dive.waterTemp }} </p>
+								</td>								
 							</tr>
 							<tr>
 								<td class="fieldTitle">
-									<h1> Air Temp. </h1>
+									<b> Max Depth </b>
 								</td>
 								<td class="fieldData">
-									<p> {{ dive.airTemp }} </p>
+									<p> {{ dive.depth }} </p>
 								</td>
-							</tr>
-							<tr>
 								<td class="fieldTitle">
-									<h1> Water Temp. </h1>
+									<b> Buddy </b>
 								</td>
 								<td class="fieldData">
-									<p> {{ dive.waterTemp }} </p>
-								</td>
+									<p> {{ dive.buddy }} </p>
+								</td>								
 							</tr>
 							<tr>
 								<td class="fieldTitle">
-									<h1> Buddy </h1>
+									 <b> Duration </b>
 								</td>
 								<td class="fieldData">
-									<p> {{ dive.buddy }} </p>
+									<p> {{ dive.duration }} </p>
 								</td>
-							</tr>
-							<tr>
 								<td class="fieldTitle">
-									<h1> Divemaster </h1>
+									<b> Dive Master </b>
 								</td>
 								<td class="fieldData">
 									<p> {{ dive.divemaster }} </p>
-								</td>
+								</td>								
 							</tr>
-							</tbody>
 						</table>
+				</div>
+
+
+				<div class="diveProfile" id="dive_{{ dive.id }}">
+
+				</div>
+
+				<div class="notesSection">
 						<table class="notes_table_class">
 							<tbody>
 							<tr>
 								<td class="fieldTitle">
-									<h1> Notes </h1>
+									<b>&nbsp;Notes </b>
 								</td>
 							</tr>
 							<tr>
 								<td class="fieldData">
-									<div class="textArea">
-										<p> {{ dive.notes|safe }} </p>
+									<input type="hidden" id="dive_notes_hidden" name="dive_notes_hidden" value="{{ dive.notes|safe }}" />
+									<div class="textArea" id="dive_notes_display" >
+										
 									</div>
 								</td>
 							</tr>
 							</tbody>
 						</table>
-					</div>
-				</div>
-			</div>
-		</div>
+				</div>  <!-- notesSection -->
+
+<script>
+    var textStr = document.getElementById('dive_notes_hidden').value;
+    var x, h, tf, y;
+    
+    x = textStr.length; // number of characters in the string
+	y = 0;
+	if ((x < 2800) & (x > 1400)) y = 0.0023 * (x-1400)/1400;
+	if (x > 2800) y = 0.0024;
+    
+    h = (0.0000015*x*x) - (0.014-y)*x + 31;
+    h = Math.floor(h);
+    
+	tf = 9 * {{template_options.font_size}};
+    if (h > tf) h = tf;  // Set template-options fontsize as maximum for notes
+	if (h < 9) h = 9;	 // Set minimum font size for notes = 9px. 
+
+     document.getElementById('dive_notes_display').innerHTML = textStr;
+     document.getElementById('dive_notes_display').style.fontSize = h;
+</script>
```

Comment:
the `JavaScript` block is whitespace damaged.
please have a look at the file `CodingStyle` and follow the guide for C/C++ here.

keywords:
- indentation
- spaces around operators

frankly, i don't like this logic. like i've said on the mailing list already. it's not portable and might not work if the user picks a very weird wide font for instance. i would make it conditional for the font families we are sure it works - e.g.:

```
if (template_options.font === "somefont1" || template_options.font === "somefont2") {
    // calculation code
}
```
and it will still remain a hack...i don't think we should include this in an official template distribution as is, because i think JavaScript people will laugh at us. but...if people want it, we might as well include it - i.e. i don't want to get in the way of users and developers... :)

interesting read about relative font sizes:
https://stackoverflow.com/questions/16056591/font-scaling-based-on-width-of-container